pkg/resourcekeeper/gc.go

```diff
@@ -103,11 +106,14 @@ func newGCConfig(options ...GCOption) *gcConfig {
 // NOTE: Mark Stage will only work when Workflow succeeds. Check/Finalize Stage will always work.
 //
 //	For one single application, the deletion will follow Mark -> Finalize -> Sweep
-func (h *resourceKeeper) GarbageCollect(ctx context.Context, options ...GCOption) (finished bool, waiting []v1beta1.ManagedResource, err error) {
```

Comment:
Let's do not change the signature of the function, `phase` parameter is not necessary here. You can set up a GCOption to carry the phase parameter and pass it to the GCConfig.

Reply:
In that way, you will only need to handle additional options instead of changing every "GarbageCollect" call.

Reply:
I know what you mean, but

EnableMarkStageGCOption takes effect when the following two conditions are met:
1. garbageCollectPolicy.ContinueOnWorkflowFail == true
2. workflow phase == WorkflowFailed
GarbageCollectPolicy is a private property of resourcekeeper, and it is better to put the logic into resourcekeeper.GarbageCollect() is better. 

If want to put the logic outside, need to make garbageCollectPolicy a public property, or add additional methods

Reply:
Yes, I originally found this problem so I think passing workflow phase inside to GarbageCollect function is necessary for now. But we can pass it inside through a GarbageCollectOption, instead of adding another typed parameter in the function inputs.